tests/network/vmi_networking.go

```diff
@@ -940,6 +940,70 @@ var _ = SIGDescribe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com]
 			)
 		})
 
+		Context("With a dedicated migration network", func() {
+			BeforeEach(func() {
+				virtClient, err = kubecli.GetKubevirtClient()
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Creating the Network Attachment Definition")
+				nad := tests.GenerateMigrationCNINetworkAttachmentDefinition()
+				_, err = virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(flags.KubeVirtInstallNamespace).Create(context.TODO(), nad, metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Failed to create the Network Attachment Definition")
+
+				// Saving the list of virt-handler pods prior to changing migration settings, see comment below.
+				listOptions := metav1.ListOptions{LabelSelector: v1.AppLabel + "=virt-handler"}
+				virtHandlerPods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(), listOptions)
+				Expect(err).ToNot(HaveOccurred(), "Failed to list the virt-handler pods")
+
+				By("Setting it as the migration network in the KubeVirt CR")
+				tests.SetDedicatedMigrationNetwork(nad.Name)
+
+				// FIXME?: By design, changing migration settings trigger a re-creation of the virt-handler pods, amongst other things.
```

Comment:
I think it could be done through conditions on the `kv` but I am also fine with your approach. I would just ask that we move the `Eventually` block into `SetDedicatedMigrationNetwork` if we ever going to use it again.